legacy-app/src/app/modules/event-templates/components/event-form-dialog/event-form-dialog.component.ts

```diff
@@ -55,6 +55,18 @@ export class EventFormDialogComponent {
     }
   }
 
+  async ngOnInit() {
+    this.dialogForm
+      .get('shouldBeReportedToInsurance')
+      ?.valueChanges.subscribe((shouldBeReportedToInsurance) => {
```

Comment:
Please make sure to clean up this subscription once the component is destroyed to prevent memory leaks